From db7f8c2ad02db911b75a9b73fd7e9bf7ba1f40c3 Mon Sep 17 00:00:00 2001
From: Zaikuo Wang <zaikuo.wang@intel.com>
Date: Thu, 7 May 2015 12:10:48 +0800
Subject: [PATCH 384/462] atomisp2:correct pipe setting when raw image capture

we should use copy_mode to decide if setting pipe_id
to copy pipe in qbuffers to css; this make sense more
because copy_mode flag has been set correctly
when atomisp_set_fmt

Change-Id: Ic2608c7b2d26e1264ba82823c86cf8eef205c02c
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-35328
Signed-off-by: Zaikuo Wang <zaikuo.wang@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/54947
Reviewed-by: Tian Shu Qiu <tian.shu.qiu@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: IMIN (SC)
Approver: Tian Shu Qiu <tian.shu.qiu@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c     | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
index 29fea23..5da32bb 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
@@ -468,7 +468,7 @@ int atomisp_qbuffers_to_css(struct atomisp_sub_device *asd)
 	}
 
 #ifdef ISP2401_NEW_INPUT_SYSTEM
-	if (asd->isp->inputs[asd->input_curr].type == SOC_CAMERA || raw_mode) {
+	if (asd->copy_mode) {
 		css_capture_pipe_id = CSS_PIPE_ID_COPY;
 		css_preview_pipe_id = CSS_PIPE_ID_COPY;
 		css_video_pipe_id = CSS_PIPE_ID_COPY;
-- 
1.9.1

